src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/RateAggregator.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.Diagnostics.Metrics
+{
+    internal class RateSumAggregator : Aggregator
+    {
+        private double _prevSum;
+        private double _sum;
+
+        public override void Update(double value)
+        {
+            lock (this)
+            {
+                _sum += value;
+            }
+        }
+
+        public override IAggregationStatistics Collect()
+        {
+            lock (this)
+            {
+                RateStatistics? stats = default;
+                double delta = _sum - _prevSum;
+                stats = new RateStatistics(delta);
+                _prevSum = _sum;
```

Comment:
if you reset _sum to zero on Collect(), then you don't need to do the delta calculation here, right?

Reply:
yep! I was doing something a little more complex before that made the 2nd sum relevant but you are right it is no longer necessary.